simplekv/net/botostore.py

```diff
@@ -105,6 +105,12 @@ def _open(self, key):
         except (BotoClientError, BotoServerError) as e:
             raise IOError(str(e))
 
+    def _copy(self, source, dest):
+        if not self._has_key(source):
```

Comment:
Doing the check independent of the copy can be subject to race conditions. Would it be possible to be optimistic and handle the resulting StorageResponseError like in the other API methods here instead?